command/format/diagnostic.go

```diff
@@ -100,18 +101,14 @@ func Diagnostic(diag tfdiags.Diagnostic, sources map[string][]byte, color *color
 					continue
 				}
 				beforeRange, highlightedRange, afterRange := lineRange.PartitionAround(highlightRange)
-				if highlightedRange.Empty() {
-					fmt.Fprintf(&buf, "%4d: %s\n", lineRange.Start.Line, sc.Bytes())
-				} else {
-					before := beforeRange.SliceBytes(src)
-					highlighted := highlightedRange.SliceBytes(src)
-					after := afterRange.SliceBytes(src)
-					fmt.Fprintf(
-						&buf, color.Color("%4d: %s[underline]%s[reset]%s\n"),
-						lineRange.Start.Line,
-						before, highlighted, after,
-					)
-				}
+				before := beforeRange.SliceBytes(src)
+				highlighted := highlightedRange.SliceBytes(src)
+				after := afterRange.SliceBytes(src)
+				fmt.Fprintf(
+					&buf, color.Color("%4d: %s[underline]%s[reset]%s\n"),
```

Comment:
I guess this works without the `if highlightedRange.Empty()` test because in that case everything will be in `before` and both `highlighted` and `after` are empty? I think perhaps when I originally wrote this I was trying to avoid printing out an underline escape code immediately followed by a reset escape code, but that doesn't really matter in practice and I agree this way seems more straightforward.

Reply:
The `highlightedRange.Empty()` check is still in this function, it was moved back to its original location:

https://github.com/hashicorp/terraform/blob/d472568d0e5714693025088169e3f4dccbc3131b/command/format/diagnostic.go#L69-L72

I'll admit I didn't look too closely at the rest of the logic - if something looks off to you I can take a closer look.

Reply:
Ahh, right... I forgot that there's the fixup step above where it turns zero-length highlights into one-byte highlights so that there's always _something_ to underline. :+1: